packages/@aws-cdk/pipelines/README.md

```diff
@@ -62,6 +107,524 @@ class MyApplication extends Stage {
   }
 }
 
+// In your main file
+new MyPipelineStack(app, 'PipelineStack', {
+  env: {
+    account: '123456789012',
+    region: 'eu-west-1',
+  }
+});
+```
+
+The pipeline is **self-mutating**, which means that if you add new
+application stages in the source code, or new stacks to `MyApplication`, the
+pipeline will automatically reconfigure itself to deploy those new stages and
+stacks.
+
+(Note that have to *bootstrap* all environments before the above code
+will work, see the section **CDK Environment Bootstrapping** below).
+
+## CDK Versioning
+
+This library uses prerelease features of the CDK framework, which can be enabled
+by adding the following to `cdk.json`:
+
+```js
+{
+  // ...
+  "context": {
+    "@aws-cdk/core:newStyleStackSynthesis": true
+  }
+}
+```
+
+## Provisioning the pipeline
+
+To provision the pipeline you have defined, making sure the target environment
+has been bootstrapped (see below), and then executing deploying the
+`PipelineStack` *once*. Afterwards, the pipeline will keep itself up-to-date.
+
+> **Important**: be sure to `git commit` and `git push` before deploying the
+> Pipeline stack using `cdk deploy`!
+>
+> The reason is that the pipeline will start deploying and self-mutating
+> right away based on the sources in the repository, so the sources it finds
+> in there should be the ones you want it to find.
+
+Run the following commands to get the pipeline going:
+
+```console
+$ git commit -a
+$ git push
+$ cdk deploy PipelineStack
+```
+
+Administrative permissions to the account are only necessary up until
+this point. We recommend you shed access to these credentials after doing this.
+
+### Working on the pipeline
+
+The self-mutation feature of the Pipeline might at times get in the way
+of the pipeline development workflow. Each change to the pipeline must be pushed
+to git, otherwise, after the pipeline was updated using `cdk deploy`, it will
+automatically revert to the state found in git.
+
+To make the development more convenient, the self-mutation feature can be turned
+off temporarily, by passing `selfMutation: false` property, example:
+
+```ts
+// Modern API
+const pipeline = new CodePipeline(this, 'Pipeline', {
+  selfMutation: false,
+  ...
+});
+
+// Original API
```

Comment:
I don't think we need to keep the documentation of the legacy API here. If you think it's valuable, I would add a section at the end of the README "Migrating from the beta API" or something like that.